engine/src/flutter/impeller/geometry/matrix.cc

```diff
@@ -358,6 +360,75 @@ std::optional<MatrixDecomposition> Matrix::Decompose() const {
   return result;
 }
 
+std::optional<std::pair<Scalar, Scalar>> Matrix::GetScales2D() const {
+  if (HasPerspective2D()) {
+    return {};
```

Comment:
Did you mean `std::nullopt` instead? Optionals get tricky when the `value_type` has a default initializer.

Reply:
(They had been passing the unit tests, so the right thing should have been happening, but I made it more explicit...)